frame/recovery/src/benchmarking.rs

```diff
@@ -0,0 +1,384 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use crate::Pallet;
+use frame_benchmarking::{account, benchmarks, whitelisted_caller};
+use frame_support::traits::{Currency, Get};
+use frame_system::RawOrigin;
+use sp_runtime::traits::{BlockNumberProvider, Bounded};
+
+const SEED: u32 = 0;
+
+fn assert_last_event<T: Config>(generic_event: <T as Config>::Event) {
+	frame_system::Pallet::<T>::assert_last_event(generic_event.into());
+}
+
+fn block_number<T: Config>() -> T::BlockNumber {
+	frame_system::Pallet::<T>::current_block_number()
+}
+
+fn get_total_deposit<T: Config>(
+	bounded_friends: &FriendsOf<T>,
+) -> Option<<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance>
+{
+	let friend_deposit = T::FriendDepositFactor::get()
+		.checked_mul(&bounded_friends.len().saturated_into())
+		.unwrap();
+
+	T::ConfigDepositBase::get().checked_add(&friend_deposit)
+}
+
+fn generate_friends<T: Config>(num: u32) -> Vec<<T as frame_system::Config>::AccountId> {
+	// Create friends
+	let mut friends = vec![];
+	for i in 0..num {
+		friends.push(account("friend", i, SEED));
+	}
+	// Sort
+	friends.sort();
+
+	for friend in 0..friends.len() {
+		// Top up accounts of friends
+		T::Currency::make_free_balance_be(
+			&friends.get(friend).unwrap(),
+			BalanceOf::<T>::max_value(),
+		);
+	}
+
+	friends
+}
+
+fn add_caller_and_generate_friends<T: Config>(
+	caller: T::AccountId,
+	num: u32,
+) -> Vec<<T as frame_system::Config>::AccountId> {
+	// Create friends
+	let mut friends = generate_friends::<T>(num - 1);
+
+	T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
+
+	friends.push(caller);
+
+	// Sort
+	friends.sort();
+
+	friends
+}
+
+fn insert_recovery_account<T: Config>(caller: &T::AccountId, account: &T::AccountId) {
+	T::Currency::make_free_balance_be(&account, BalanceOf::<T>::max_value());
+
+	let n = T::MaxFriends::get();
+
+	let friends = generate_friends::<T>(n);
+	let delay_period = block_number::<T>();
+
+	let bounded_friends: FriendsOf<T> = friends.try_into().unwrap();
+
+	// Get deposit for recovery
+	let total_deposit = get_total_deposit::<T>(&bounded_friends).unwrap();
+
+	let recovery_config = RecoveryConfig {
+		delay_period,
+		deposit: total_deposit,
+		friends: bounded_friends,
+		threshold: n as u16,
+	};
+
+	// Reserve deposit for recovery
+	T::Currency::reserve(&caller, total_deposit).unwrap();
+
+	<Recoverable<T>>::insert(&account, recovery_config);
+}
+
+benchmarks! {
+	as_recovered {
+		let caller: T::AccountId = whitelisted_caller();
+		let recovered_account: T::AccountId = account("recovered_account", 0, SEED);
+		let call: <T as Config>::Call = frame_system::Call::<T>::remark { remark: vec![] }.into();
+
+		Proxy::<T>::insert(&caller, &recovered_account);
+	}: _(
+		RawOrigin::Signed(caller),
+		recovered_account,
+		Box::new(call)
+	)
```

Comment:
you could use `remark_with_event` here and assert it actually executed.

Reply:
I am unfamiliar with this, but I will give it a try.

Reply:
I am having issues getting this to work, I wouldn't mind some help.

Reply:
do you have a commit of your attempt?

Reply:
This is what I did

```
	as_recovered {
		let caller: T::AccountId = whitelisted_caller();
		let recovered_account: T::AccountId = account("recovered_account", 0, SEED);
		let call: <T as Config>::Call = frame_system::Call::<T>::remark_with_event { remark: vec![] }.into();
		let call_hash = <T as frame_system::Config>::Hashing::hash_of(&call);

		Proxy::<T>::insert(&caller, &recovered_account);
	}: _(
		RawOrigin::Signed(caller.clone()),
		recovered_account,
		Box::new(call)
	) verify {
		assert_last_event(
			frame_system::<T>::Event::Remarked { sender: caller, hash: call_hash }
		);
	}
```